### PR TITLE
fix: hide pattern settings button if can't edit

### PR DESCRIPTION
**Pull request by brandonreid**

### CODE REVIEW

File: knapsack/src/client/layouts/page-with-sidebar.tsx [hunk 1]
```diff
@@ -88,7 +89,7 @@ const PageWithSidebar: React.FC<Props> = ({
           {children}
         </main>
       </ErrorCatcher>
-      {slottedDetails && (
+      {canEdit && slottedDetails && (
```
Comment (EvanLovely) [inline knapsack/src/client/layouts/page-with-sidebar.tsx]: This is the wrong spot for the `canEdit` - This is basically "right sidebar". Currently we are only using the Pattern Settings in the right sidebar, but in the future we'll use other things as well.

Comment (EvanLovely) [reply knapsack/src/client/layouts/page-with-sidebar.tsx]: We could probably hide the collapse button if `!slottedDetails`

Comment (EvanLovely) [reply knapsack/src/client/layouts/page-with-sidebar.tsx]: The pattern view page passes in `slottedDetails` here: 
https://github.com/basaltinc/knapsack/blob/af90ceb64c8f89b8cc6ecc896e431ffcc7a0ccf2/knapsack/src/client/pages/pattern-view/pattern-view-page.tsx#L152-L161

Comment (brandonreid) [reply knapsack/src/client/layouts/page-with-sidebar.tsx]: @EvanLovely Resolved. Moved the button into the `pattern-settings`, the whole chunk there is hidden with the same method. Also makes the button stay fixed when scrolling the main stage, just like the left-nav collapse btn.

File: knapsack/src/client/pages/pattern-view/components/pattern-settings/pattern-settings.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /* eslint-disable no-else-return */
```
Comment (brandonreid) [inline knapsack/src/client/pages/pattern-view/components/pattern-settings/pattern-settings.tsx]: This file is just adding the button, then wrapping all the content in an `__inner` div for scroll behavior.

Comment (EvanLovely) [reply knapsack/src/client/pages/pattern-view/components/pattern-settings/pattern-settings.tsx]: Thanks for the heads up :+1:

**Comment by EvanLovely** [comment]
Whoops; merged that into `next`...

**Pull request merged**